inference/__init__.py

```diff
@@ -0,0 +1,492 @@
+    DENSE_ONLY_FP16 = DENSE_ONLY | FP16
+    SPARSE_ALGO0_FP16 = SPARSE_ALGORITHM0 | FP16
+    SPARSE_ALGO0_FP32 = SPARSE_ALGORITHM0 | FP32
+    SPARSE_NAIVE_FP16 = SPARSE_ALGORITHM_NAIVE | FP16
+    SPARSE_SHARED_FP16 = SPARSE_SHARED_ALGORITHM | FP16
+    SPARSE_SINGLE_FP16 = SPARSE_SINGLE_ALGORITHM | FP16
+    SPARSE_SHARED_BASELINE_FP16 = SPARSE_SHARED_BASELINE_ALGORITHM | FP16
+    SPARSE_CUSPARSE_FP16 = SPARSE_CUSPARSE_ALGORITHM | FP16
+    SPARSE_FUSED_FP16 = SPARSE_FUSED_ALGORITHM | FP16
+    SPARSE_FUSED_FP32 = SPARSE_FUSED_ALGORITHM | FP32
+    SPARSE_FUSED_FP32_ASYNC = SPARSE_FUSED_ALGORITHM | FP32 | IS_ASYNC
+    SPARSE_NAIVE_FP32 = SPARSE_ALGORITHM_NAIVE | FP32
+    TORCH_FP16 = TORCH | FP16
+    TORCH_FP32 = TORCH | FP32
+    SPARSE_MIXTURE_FP16 = SPARSE_SHARED_MIXTURE_ALGORITHM | FP16
+    SPARSE_MIXTURE_FP32 = SPARSE_SHARED_MIXTURE_ALGORITHM | FP32
+
+    def pretty(self):
+        if self.value == FeatureFlag.TORCH_FP16:
+            return 'Torch FP16'
+        elif self.value == FeatureFlag.SPARSE_MIXTURE_FP16:
+            return 'SPQR Kernel'
+        elif self.value == FeatureFlag.SPARSE_MIXTURE_FP32:
+            return 'SPQR Kernel (FP32)'
+        elif self.value == FeatureFlag.DENSE_ONLY_FP16:
+            return 'SPQR Kernel Dense Only'
+        elif self.value == FeatureFlag.DENSE_ONLY_FP32:
+            return 'SPQR Kernel Dense Only (FP32)'
+        elif self.value == FeatureFlag.SPARSE_SHARED_FP16:
+            return 'Sparse With Sharing'
+        elif self.value == FeatureFlag.SPARSE_SHARED_BASELINE_FP16:
+            return 'Sparse Without Sharing'
+        elif self.value == FeatureFlag.SPARSE_FUSED_FP16:
+            return 'Sparse Fused FP16'
+        elif self.value == FeatureFlag.SPARSE_FUSED_FP32:
+            return 'Sparse Fused FP32'
+        elif self.value == FeatureFlag.SPARSE_FUSED_FP32_EXPERIMENTAL:
+            return 'Sparse Fused FP32 Experimental'
+        else:
+            raise 'Pretty print not found for value {self.value}'
+
+
+def spqr_mul(spqr_device: SPQRModule, x, y, feature_flag: FeatureFlag):
+    spqr_cuda.spqr_mul(
+        spqr_device.m,
+        spqr_device.n,
+        spqr_device.bits,
+        spqr_device.beta1,
+        spqr_device.beta2,
+        spqr_device.buff0,
+        spqr_device.row_offsets,
+        spqr_device.col_vals,
+        spqr_device.nnz,
+        x,
+        y,
+        int(feature_flag)
+    )
+
+
+def torch_mul_timer(deq_w, x, num_runs):
+    if len(deq_w.shape) == 1:
+        n = x.shape[0]
+        m = deq_w.shape[0] // n
+    else:
+        m, n = deq_w.shape
+
+    assert (n == x.shape[0])
+
+    runs = torch.empty(num_runs).cpu().float()
+
+    y = torch.zeros(m, dtype=x.dtype, device=x.device)
+
+    for i in range(num_runs):
+        y = torch.zeros_like(y)
+        spqr_cuda.torch_mul_fp16(m, n, deq_w, x, y, runs[i])
+
+    return y, runs
+
+
+def spqr_mul_timer(spqr_device: SPQRModule, x, feature_flag: FeatureFlag, num_runs):
```

Comment:
Do the timer functions should be in __init__.py?

Reply:
But it's best to keep them in benchmark scope until this I get around implementing this.